docker-compose.yml

```diff
@@ -0,0 +1,19 @@
+version: '3.2'
+services:
+  pl:
+    build:
+      context: .
+    ports:
+      - 3000:3000
+    volumes:
+      - ./testCourse:/course
+      - /var/run/docker.sock:/var/run/docker.sock
+      - ${HOME}/pl_ag_jobs:/jobs
+    container_name: pl
+    environment:
+      - HOST_JOBS_DIR=${HOME}/pl_ag_jobs
+
+    # Starts the server with a different starting command
+    # Replaces /PrairieLearn/docker/init.sh
+    # This isn't functional on windows machines
+    # command: /PrairieLearn/docker/start_workspace.sh
```

Comment:
```suggestion
```

Hmm I can't get this to work on linux either: `open terminal failed: not a terminal`. Maybe some issue with the script using tmux?

Also now that I see the docs below, I'm realizing this is more for instructors doing course dev, so maybe we should remove this anyway. It might make sense to have another yml for PL dev that includes `command`, `/PrairieLearn` mount, etc.

Reply:
It might be. I'm not the most familiar with tmux, so I wasn't sure if it was just a windows config issue.

It it still useful to add this PR in (as in this `docker-compose`, albiet with the command removed) or not?

Reply:
Personally I still think this docker-compose (sans `command`) is useful for instructors, but we can see what others think in the dev meeting. Just seems like a simple `docker-compose up` with an editable yml is much easier to digest than a huge shell command.

Maybe we could have a separate yml for PL dev work with stuff like `command`, `-v ./:/PrairieLearn`, etc.

Reply:
This docker file builds the pl server from the given dockerfile, so we shouldn't have to copy over the entire root dir into `/PrairieLearn`, but agreed that this is easier to understand than the shell.

I unfortunately won't be able to make the dev meeting today, but I'll be here and in the dev channel, so keep me up to date.

Reply:
> This docker file builds the pl server from the given dockerfile, so we shouldn't have to copy over the entire root dir into `/PrairieLearn`

I usually mount `./:/PrairieLearn` so I can hot-reload the node server itself (not the whole container) when the source changes, but maybe there's a better way to do that with docker-compose?